test/framework/ownerreference_helpers.go

```diff
@@ -0,0 +1,308 @@
+}
+
+// Kind and GVK for types in the Docker infrastructure package.
+var (
+	dockerMachineKind         = "DockerMachine"
+	dockerMachineTemplateKind = "DockerMachineTemplate"
+	dockerClusterKind         = "DockerCluster"
+	dockerClusterTemplateKind = "DockerClusterTemplate"
+)
+
+// DockerInfraOwnerReferenceAssertions maps Docker Infrastructure types to functions which return an error if the passed
+// OwnerReferences aren't as expected.
+var DockerInfraOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
+	dockerMachineKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{machineGVK})
+	},
+	dockerMachineTemplateKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK}, []schema.GroupVersionKind{clusterClassGVK})
+	},
+	dockerClusterKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK})
+	},
+	dockerClusterTemplateKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterClassGVK})
+	},
+}
+
+func hasExactOwnersByGVK(refList []metav1.OwnerReference, wantGVKs []schema.GroupVersionKind) error {
+	refGVKs := []schema.GroupVersionKind{}
+	for _, ref := range refList {
+		refGVK, err := ownerRefGVK(ref)
+		if err != nil {
+			return err
+		}
+		refGVKs = append(refGVKs, refGVK)
+	}
+	sort.SliceStable(refGVKs, func(i int, j int) bool {
+		return refGVKs[i].String() > refGVKs[j].String()
+	})
+	sort.SliceStable(wantGVKs, func(i int, j int) bool {
+		return wantGVKs[i].String() > wantGVKs[j].String()
+	})
+	if !reflect.DeepEqual(wantGVKs, refGVKs) {
+		return fmt.Errorf("wanted %v, actual %v", wantGVKs, refGVKs)
+	}
+	return nil
+}
+
+func hasOneOfExactOwnersByGVK(refList []metav1.OwnerReference, possibleGVKS ...[]schema.GroupVersionKind) error {
+	var allErrs []error
+	for _, wantGVK := range possibleGVKS {
+		err := hasExactOwnersByGVK(refList, wantGVK)
+		if err != nil {
+			allErrs = append(allErrs, err)
+			continue
+		}
+		return nil
+	}
+	return kerrors.NewAggregate(allErrs)
+}
+
+func ownerRefGVK(ref metav1.OwnerReference) (schema.GroupVersionKind, error) {
+	refGV, err := schema.ParseGroupVersion(ref.APIVersion)
+	if err != nil {
+		return schema.GroupVersionKind{}, err
+	}
+	return schema.GroupVersionKind{Version: refGV.Version, Group: refGV.Group, Kind: ref.Kind}, nil
+}
+
+func setClusterPause(ctx context.Context, cli client.Client, clusterKey types.NamespacedName, value bool) {
+	cluster := &clusterv1.Cluster{}
+	Expect(cli.Get(ctx, clusterKey, cluster)).To(Succeed())
+
+	unpausePatch := client.RawPatch(types.MergePatchType, []byte(fmt.Sprintf("{\"spec\":{\"paused\":%v}}", value)))
+	Expect(cli.Patch(ctx, cluster, unpausePatch)).To(Succeed())
+}
+
+func annotateClusterClass(ctx context.Context, cli client.Client, clusterKey types.NamespacedName) {
```

Comment:
```suggestion
func forceClusterClassReconcile(ctx context.Context, cli client.Client, clusterKey types.NamespacedName) {
```
or
triggerClusterClassReconcile